kernel/dosfns.c

```diff
@@ -255,6 +255,54 @@ long DosRWSft(int sft_idx, size_t n, __XFAR(void)bp, int mode)
 }
 
 COUNT SftSeek(int sft_idx, LONG new_pos, unsigned mode)
+{
+  unsigned long result;
+  return SftSeek2(sft_idx, new_pos, mode, &result);
+}
+
+static unsigned try_long_compat_remote_seek(sft FAR * s, LONG new_pos,
+	unsigned mode, unsigned long * p_result)
+{
+  if (0 == (s->sft_flags & SFT_FSHARED))
+    return 0;
+
+  struct seek_struct_S {
+    uint32_t member[2];
+  };
+  struct seek_struct_S seek_struct;
+  struct seek_struct_S FAR * ssp;
+  seek_struct.member[0] = new_pos;
+  if (mode == SEEK_SET)
+    seek_struct.member[1] = 0;
+  else if (0 == (new_pos & 0x80000000))
+    seek_struct.member[1] = 0;
+  else
+    seek_struct.member[1] = 0xFFFFffff;
+
+  iregs regs = {};
```

Comment:
Would be better to move that up to
the declarations.

Reply:
Done.